src/app/domain/backup/services/manageRemoteBackupConfig.ts

```diff
@@ -0,0 +1,148 @@
+import DeviceInfo from 'react-native-device-info'
+
+import { BackupedMedia, RemoteBackupConfig } from '/app/domain/backup/models'
+
+import type CozyClient from 'cozy-client'
+
+import { buildFilesQuery, FilesQueryResult } from '/app/domain/backup//queries'
+
+const DOCTYPE_APPS = 'io.cozy.apps'
+const DOCTYPE_FILES = 'io.cozy.files'
+const BACKUP_REF = `io.cozy.apps/photos/mobile`
+const BACKUP_ROOT_PATH = '/Sauvegardé depuis mon mobile'
+
+interface BackupFolder {
+  _id: string
+  attributes: {
+    name: string
+    path: string
+    metadata?: {
+      backupDeviceIds: string[]
+    }
+  }
+}
+
+const isInTrash = (path: string): boolean => {
+  return path.startsWith('/.cozy_trash')
+}
+
+export const fetchRemoteBackupConfigs = async (
+  client: CozyClient
+): Promise<RemoteBackupConfig[]> => {
+  const { included } = await client.collection(DOCTYPE_FILES).findReferencedBy({
+    _type: DOCTYPE_APPS,
+    _id: BACKUP_REF
+  })
+
+  const backupFolders = included as BackupFolder[]
+
+  const remoteBackupConfigs: RemoteBackupConfig[] = backupFolders
+    .filter(folder => !isInTrash(folder.attributes.path))
+    .map(backupFolder => ({
+      backupFolder: {
+        id: backupFolder._id,
+        name: backupFolder.attributes.name,
+        path: backupFolder.attributes.path
+      },
+      backupDeviceIds: backupFolder.attributes.metadata?.backupDeviceIds ?? []
+    }))
+
+  return remoteBackupConfigs
+}
+
+export const isRemoteBackupConfigFromDevice = (
+  remoteBackupConfig: RemoteBackupConfig,
+  deviceId: string
+): boolean => {
+  return remoteBackupConfig.backupDeviceIds.includes(deviceId)
+}
+
+export const fetchDeviceRemoteBackupConfig = async (
+  client: CozyClient
+): Promise<RemoteBackupConfig | undefined> => {
+  const remoteBackupConfigs = await fetchRemoteBackupConfigs(client)
+  const deviceId = await getDeviceId()
+
+  const remoteBackupConfig = remoteBackupConfigs.find(remoteBackupConfig =>
+    isRemoteBackupConfigFromDevice(remoteBackupConfig, deviceId)
+  )
+
+  return remoteBackupConfig
+}
+
+export const createRemoteBackupFolder = async (
+  client: CozyClient
+): Promise<RemoteBackupConfig> => {
+  const deviceName = await getDeviceName()
+  const deviceId = await getDeviceId()
+
+  const {
+    data: { _id: backupRootId }
+  } = await client
+    .collection(DOCTYPE_FILES)
+    .createDirectoryByPath(BACKUP_ROOT_PATH)
```

Comment:
Can't you use `client.save()` directly? You should avoid calling `collection()`

Reply:
nit: Oh yeah didn't see that it was `createDirectoryByPath`. Maybe we can update CozyClient to accept that one: 

```
 async create(attributes) {
    if (attributes.type === 'directory') {
      return this.createDirectory(attributes)
    } else {
      const { data, ...createFileOptions } = attributes
      return this.createFile(data, createFileOptions)
    }
  }

async createDirectory(...){
if(params.path){
  createDirectoryByPath()
}
...
}
```

Reply:
Ok! I prefer to keep it that way for the moment. Take time to modify cozy-client and createDirectoryByPath is kind of very explicit.